json_encoder.go

```diff
@@ -104,6 +105,18 @@ func (enc *jsonEncoder) AddInt64(key string, val int64) {
 // grade-school notation otherwise).
 func (enc *jsonEncoder) AddFloat64(key string, val float64) {
 	enc.addKey(key)
+	if math.IsNaN(val) {
+		enc.bytes = append(enc.bytes, `"NaN"`...)
+		return
+	}
+	if math.IsInf(val, 1) {
+		enc.bytes = append(enc.bytes, `"+Inf"`...)
+		return
+	}
+	if math.IsInf(val, -1) {
```

Comment:
Right, but we'll still need to distinguish +Inf from -Inf. Is

```go
if math.IsInf(val, 0) {
   if val > 0 {
       // +Inf
    } else {
        // -Inf
    }
}
```
better than two `math.IsInf` checks?